clients/pkg/promtail/scrapeconfig/scrapeconfig.go

```diff
@@ -246,6 +247,30 @@ type WindowsEventsTargetConfig struct {
 	Labels model.LabelSet `yaml:"labels"`
 }
 
+type AzurelogTargetConfig struct {
+	// Labels optionally holds labels to associate with each log line.
+	Labels model.LabelSet `yaml:"labels"`
+
+	// UseIncomingTimestamp sets the timestamp to the incoming messages
+	// timestamp if it's set.
+	UseIncomingTimestamp bool `yaml:"use_incoming_timestamp"`
+
+	// Event Hub Topics to consume (Required).
+	Topics []string `yaml:"topics"`
+
+	// Event Hub ConnectionString for authentication on Azure Cloud (Required).
+	ConnectionString string `yaml:"connection_string"`
+
+	// The list of brokers to connect to Event Hub (Required).
+	Brokers []string `yaml:"brokers"`
+
+	// The consumer group id.
+	GroupID string `yaml:"group_id"`
+
+	// Do not ignore messages that doesn't match schema for Azure resource logs
+	AllowCustomPayload bool `yaml:"allow_custom_payload"`
+}
```

Comment:
I wonder if at this level we should try to use as much Event Hub terminology as possible to avoid potential confusion and limit changes in the event we swap out the kafka consumer for the go azure client. I think this would only be two renames, `Topics` -> `EventHubs` and `Brokers`- > `ServiceBusEndpoint`? (I don't actually remember what this one is

Reply:
👍 that makes sense!

Reply:
Done 👍